platforms/core-configuration/file-collections/src/main/java/org/gradle/api/internal/file/collections/DefaultConfigurableFileCollection.java

```diff
@@ -203,14 +204,39 @@ public Set<Object> getFrom() {
     public void setFromAnyValue(Object object) {
         // Currently we support just FileCollection for Groovy assign, so first try to cast to FileCollection
         FileCollectionInternal fileCollection = Cast.castNullable(FileCollectionInternal.class, Cast.castNullable(FileCollection.class, object));
+        checkNoSelfReferences(fileCollection);
```

Comment:
❓ Should we also have this check when handling assignment to CFC in Kotlin?

Reply:
We should, but it should be a more general self-reference check, outside of the scope of this PR